st2common/st2common/validators/api/reactor.py

```diff
@@ -65,11 +66,14 @@ def validate_trigger_parameters(trigger_type_ref, parameters):
     if not trigger_type_ref:
         return None
 
-    if trigger_type_ref not in SYSTEM_TRIGGER_TYPES:
-        # Not a system trigger, skip validation for now
+    trigger_type = triggers.get_trigger_type_db(trigger_type_ref)
+    if trigger_type_ref in SYSTEM_TRIGGER_TYPES:
+        parameters_schema = SYSTEM_TRIGGER_TYPES[trigger_type_ref]['parameters_schema']
```

Comment:
Thanks.

This change looks good to me now, but it's worth noting that this method is also to validate parameters when creating a rule so we also need some flag to disable validation during rule creation as well (similar to what you did with `permit_invalid_payload`).

In addition to that, we also need some tests for rule creation with invalid parameters when validation is turned on.